resources/fetching/fetchers/azure/assets_fetcher.go

```diff
@@ -119,6 +127,44 @@ func (f *AzureAssetsFetcher) Fetch(ctx context.Context, cycleMetadata cycle.Meta
 	return errAgg
 }
 
+func (f *AzureAssetsFetcher) enrichStorageAccountAssets(ctx context.Context, cycleMetadata cycle.Metadata, subscriptions map[string]governance.Subscription, assets []inventory.AzureAsset) error {
+	diagSettings, err := f.provider.ListDiagnosticSettingsAssetTypes(ctx, cycleMetadata, lo.Keys(subscriptions))
+	if err != nil {
+		return err
+	}
+
+	addUsedForActivityLogsFlag(assets, diagSettings)
+
+	return nil
+}
+
+func addUsedForActivityLogsFlag(assets []inventory.AzureAsset, diagSettings []inventory.AzureAsset) {
+	usedStorageAccountIDs := map[string]struct{}{}
+	for _, d := range diagSettings {
+		storageAccountID := strings.FromMap(d.Properties, "storageAccountId")
+		if storageAccountID == "" {
+			continue
+		}
+		usedStorageAccountIDs[storageAccountID] = struct{}{}
+	}
+
+	for i, a := range assets {
+		if a.Type != inventory.StorageAccountAssetType {
+			continue
+		}
+
+		if _, exists := usedStorageAccountIDs[a.Id]; !exists {
+			continue
+		}
+
+		if a.Extension == nil {
+			a.Extension = make(map[string]any)
+		}
+		a.Extension["usedForActivityLogs"] = true
```

Comment:
The resource is the storage account, and we enrich it with the information "_is this storage account used to store activity logs_".  

Perhaps the diagnostic settings id, and/or diagnostic settings name might be more informative as you suggested. 

In any case, I'd suggest doing it after the merge in order to be sure that the rule is included before the feature freeze. 

Then we can open a PR with this change (it will need changes in code and rego as well).

What do you think?

Reply:
> Perhaps the diagnostic settings id, and/or diagnostic settings name might be more informative as you suggested.

Yes

> In any case, I'd suggest doing it after the merge in order to be sure that the rule is included before the feature freeze.
> 
> Then we can open a PR with this change (it will need changes in code and rego as well).
> 
> What do you think?

Up to you to estimate the time it needs to be done and which release it should go to. If it is postponed for 8.13 we will need to QA it.